### ISSUE TITLE
USER ISSUE: Setting up recurring payment at registrar's desk causes irrecoverable server crash

### ISSUE THREAD

**Issue by krixi** [StrangeLoopGames/EcoIssues#20274]
**User Id: (Not logged user)**
**Version:** 0.9.2.2 beta release-149

**Steps to Reproduce:**
1. I was playing the game locally with one other person. Settings were No Collaboration, 100 days to impact. 
2. We had researched nearly all of the technologies, built the lasers and destroyed the meteor. 
3. We had NOT built the capitol or any government buildings, aside from the Registrar's desk. 
4. I was playing around in the registrar's desk and gave the other player a title, and then tried to set up a recurring payment for that title from my personal currency. 
5. The game locked up and this error pops up. 
6. Now, every time I try to load the game, this error pops, and I'm unable to continue playing this save. 

**Expected behavior:**
It probably shouldn't crash :)

**Actual behavior:**
The following error message popped up:
```
<size=150%>Connection Failed</size>

Server encountered an exception:

<size=60.00%>Exception: NullReferenceException
Message:Object reference not set to an instance of an object.
Source:Eco.Gameplay

System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.Gameplay.Economy.Jobs.RecurringTransferManager.<>c__DisplayClass8_0.<TransfersForUser>b__0(RecurringTransfer x)
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at Eco.Core.Controller.ControllerManager.PackageValue(IMvcNetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageProperty(IMvcNetClient boundClient, IController controller, ISyncPropertyInfo propertyInfo)
Outer Exceptions:
Error during packaging property 'IncomeAndExpenses' of 'Eco.Gameplay.Players.User'</size>
```

**Do you have mods installed? Does the issue happen when no mods are installed?:**
No mods are installed. 

**Please provide log files. Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Getting-the-Eco-game-log-file

Logs here: 
[eco-crash-log.txt](https://github.com/StrangeLoopGames/EcoIssues/files/5898874/eco-crash-log.txt)

Please let me know if there's anything else I can do to help, I've got 100 hours into this map 😅 🤞

**Comment by apgamingreal** [comment]
Similar issue, every time I try to join server after 9.2.2 update the following error shows up

No mods

[log_210130055410.log](https://github.com/StrangeLoopGames/EcoIssues/files/5898910/log_210130055410.log)

**Comment by WeaselDog** [comment]
Thank you for letting us know. We will try to get this fixed as soon as possible. You should have a backup save from before this happened you can restore in the meantime.

**Comment by MishkaFur** [comment]
[Player.log](https://github.com/StrangeLoopGames/EcoIssues/files/5898970/Player.log)

Able to replicate the issue, go into a registra title, go into wages, boom client is kicked and cannot rejoin even after a restart of client and server

Server console shows

![image](https://user-images.githubusercontent.com/61889138/106370996-70c20480-6357-11eb-98b8-358bc90e5fc9.png)

Dump from server, nothing in server logs
[Dump.zip](https://github.com/StrangeLoopGames/EcoIssues/files/5898971/Dump.zip)

**Comment by Venopsis** [comment]
Hi,

Also got kind of similar issue. My server crashed and not able to recover a working state without backup restoration.

Happened when I used command /objects edit title,ID to check the wage issue on this elected title (Roi).

The error was that with wage enable for type online or online/offline an amount of money must be set.

After few seconds boom the server kicked me out and nobody is able to join. Even if they were not connected at the time of the issue.

**Comment by Venopsis** [comment]
@MishkaFur Were did you get those dump files? I'll check on mine and see if I can add them to my case. Could be useful ;) Thanks!

**Comment by MishkaFur** [comment]
> 
> 
> @MishkaFur Where did you get those dump files? I'll check on mine and see if I can add them to my case. Could be useful ;) Thanks!

server folder has one called dump, you'll need to zip it as github wont allow the format to be uploaded otherwise

**Comment by krixi** [comment]
Thanks for jumping on this so quickly! I was able to find the backups and restore from a previous version. 

For anyone else reading who's curious, for me they were in:
```
Program Files (x86)/SteamLibrary/steamapps/common/Eco/Eco_Data/Server/Storage/Backup
```

It looks like the primary game data is stored in `Eco/Eco_Data/Server/Storage`, in two files (a .db and a .eco file). I copied the corrupt ones out to a safe place, then copied over them with the latest backup I could find and it seems to have worked - only lost about 20 mins of work. 

Thank you! 😁

**Comment by SlayksWood** [comment]
To full testing of this issue need to load this save file:
https://drive.google.com/file/d/1EVY8BxTGnEtJmLD-wPADLmHOWUQPrr1T/view?usp=sharing
This server with already broken system. Just download it after fixes and try to connect to server.

**Comment by Venopsis** [comment]
Hi @SlayksWood ,
Saw  your message on my case. No problem to get all this done here ;)
So you want us to download those two files, replace on server and try to connect right?
If this works, what's the next step?
Thanks!
PS: I'm not at home before later this evening (Brussels time) so I cannot test this right now.

**Comment by IcedForge** [comment]
@Venopsis Slays test comment regarding the save i believe was directed at our internal QA for further testing.

**Comment by WeaselDog** [comment]
This will be fixed in the update 9.2.3. Existing saves will be able to be recovered.

**Comment by Venopsis** [comment]
Installed the 9.2.3 right now and unfortunately this update didn't solve my issue.

```
<size=150%>Connection Failed</size>

Server encountered an exception:

<size=60.00%>Exception: NullReferenceException
Message:Object reference not set to an instance of an object.
Source:Eco.Gameplay

System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.Gameplay.Economy.Jobs.RecurringTransfer.get_SourceAccount()
   at Eco.Gameplay.Economy.Jobs.RecurringTransferManager.<>c__DisplayClass8_0.<TransfersForUser>b__0(RecurringTransfer x)
   at System.Linq.Utilities.<>c__DisplayClass1_0`1.<CombinePredicates>b__0(TSource x)
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at Eco.Core.Controller.ControllerManager.PackageValue(IMvcNetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageProperty(IMvcNetClient boundClient, IController controller, ISyncPropertyInfo propertyInfo)
Outer Exceptions:
Error during packaging property 'IncomeAndExpenses' of 'Eco.Gameplay.Players.User'</size>

Log File:
Initialize engine version: 2019.4.15f1 (fbf367ac14e9)
[Subsystems] Discovering subsystems at path F:/Steam/steamapps/common/Eco/Eco_Data/UnitySubsystems
GfxDevice: creating device client; threaded=1
Direct3D:
    Version:  Direct3D 11.0 [level 11.1]
    Renderer: NVIDIA GeForce GTX 1070 Ti (ID=0x1b82)
    Vendor:   
    VRAM:     8079 MB
    Driver:   27.21.14.6089
D3D11 device created for Microsoft Media Foundation video decoding.
<RI> Initializing input.

<RI> Input initialized.

<RI> Initialized touch support.

UnloadTime: 0.313000 ms
DontDestroyOnLoad only works for root GameObjects or components on root GameObjects. 
(Filename:  Line: 589)

WwiseUnity: Wwise(R) SDK Version 2019.2.5 Build 7349.
UnityEngine.Logger:LogFormat(LogType, String, Object[])
UnityEngine.Debug:LogFormat(String, Object[])
AkWwiseInitializationSettings:InitializeSoundEngine()
AkInitializer:OnEnable()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

WwiseUnity: Setting Plugin DLL path to: F:/Steam/steamapps/common/Eco/Eco_Data\Plugins\x86_64
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
AkCommonPlatformSettings:get_AkInitializationSettings()
AkWwiseInitializationSettings:InitializeSoundEngine()
AkInitializer:OnEnable()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

WwiseUnity: Sound engine initialized successfully.
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
AkWwiseInitializationSettings:InitializeSoundEngine()
AkInitializer:OnEnable()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

The referenced script on this Behaviour (Game Object 'Camera') is missing! 
(Filename:  Line: 334)

The referenced script on this Behaviour (Game Object 'Table') is missing! 
(Filename:  Line: 334)

DontDestroyOnLoad only works for root GameObjects or components on root GameObjects.
SteamManager:Awake()

[ line 589] 
(Filename:  Line: 589)

Ambiguous UI component types ignored: SeriesPopupUI,BaseUI,ViewEditorUI,AutoViewUI
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
UI.UIManager:Start()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

[04:05:17.355] LoginSession: connectorHandle=C0
EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
VivoxUnity.VivoxDebug:DebugMessage(Object, vx_log_level)
VivoxUnity.Private.<>c__DisplayClass89_0:<BeginLogin>b__0(IAsyncResult)
Trigger:Invoke(GameObject)
VivoxUnity.AsyncResult`1:SetComplete(T)
VivoxUnity.<>c__DisplayClass12_0:<BeginGetConnectorHandle>b__0(IAsyncResult)
Trigger:Invoke(GameObject)
VivoxUnity.AsyncResult`1:SetComplete(T)
VivoxUnity.VxClient:InstanceOnMainLoopRun(Boolean&)
MessagePump:RunOnce()
MessagePump:RunUntil(LoopDone)
VoiceManager:Update()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

DirectoryNotFoundException: Could not find a part of the path 'F:\Steam\steamapps\common\Eco\Eco_Data\Server\Storage\Backup'.
  at System.IO.__Error.WinIOError (System.Int32 errorCode, System.String maybeFullPath) [0x00000] in <00000000000000000000000000000000>:0 
  at System.IO.FileSystemEnumerableIterator`1[TSource].CommonInit () [0x00000] in <00000000000000000000000000000000>:0 
  at System.IO.FileSystemEnumerableIterator`1[TSource]..ctor (System.String path, System.String originalUserPath, System.String searchPattern, System.IO.SearchOption searchOption, System.IO.SearchResultHandler`1[TSource] resultHandler, System.Boolean checkHost) [0x00000] in <00000000000000000000000000000000>:0 
  at System.IO.Directory.InternalGetFileDirectoryNames (System.String path, System.String userPathOriginal, System.String searchPattern, System.Boolean includeFiles, System.Boolean includeDirs, System.IO.SearchOption searchOption, System.Boolean checkHost) [0x00000] in <00000000000000000000000000000000>:0 
  at System.IO.Directory.GetDirectories (System.String path) [0x00000] in <00000000000000000000000000000000>:0 
  at ServerListingLocalUI.EditBackups (System.String fileDirectory, System.String fileName, System.Action`1[T] whatToDo) [0x00000] in <00000000000000000000000000000000>:0 
  at ServerListingLocalUI.<Delete>b__19_0 () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.AndroidJavaRunnable.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at PopupDialogUI.OnYesClick () [0x00000] in <00000000000000000000000000000000>:0 
  at Steamworks.ISteamMatchmakingRulesResponse+RulesFailedToRespond.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.Events.InvokableCall.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.Events.UnityEvent.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at System.UnhandledExceptionEventHandler.Invoke (System.Object sender, System.UnhandledExceptionEventArgs e) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.StandaloneInputModule.ReleaseMouse (UnityEngine.EventSystems.PointerEventData pointerEvent, UnityEngine.GameObject currentOverGo) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.StandaloneInputModule.ProcessMousePress (UnityEngine.EventSystems.PointerInputModule+MouseButtonEventData data) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.StandaloneInputModule.ProcessMouseEvent (System.Int32 id) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.StandaloneInputModule.Process () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.EventSystem.Update () [0x00000] in <00000000000000000000000000000000>:0 
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
UnityEngine.EventSystems.ExecuteEvents:Execute(GameObject, BaseEventData, EventFunction`1)
UnityEngine.EventSystems.StandaloneInputModule:ReleaseMouse(PointerEventData, GameObject)
UnityEngine.EventSystems.StandaloneInputModule:ProcessMousePress(MouseButtonEventData)
UnityEngine.EventSystems.StandaloneInputModule:ProcessMouseEvent(Int32)
UnityEngine.EventSystems.StandaloneInputModule:Process()
UnityEngine.EventSystems.EventSystem:Update()
 
(Filename: currently not available on il2cpp Line: -1)

DirectoryNotFoundException: Could not find a part of the path 'F:\Steam\steamapps\common\Eco\Eco_Data\Server\Storage\Backup'.
  at System.IO.__Error.WinIOError (System.Int32 errorCode, System.String maybeFullPath) [0x00000] in <00000000000000000000000000000000>:0 
  at System.IO.FileSystemEnumerableIterator`1[TSource].CommonInit () [0x00000] in <00000000000000000000000000000000>:0 
  at System.IO.FileSystemEnumerableIterator`1[TSource]..ctor (System.String path, System.String originalUserPath, System.String searchPattern, System.IO.SearchOption searchOption, System.IO.SearchResultHandler`1[TSource] resultHandler, System.Boolean checkHost) [0x00000] in <00000000000000000000000000000000>:0 
  at System.IO.Directory.InternalGetFileDirectoryNames (System.String path, System.String userPathOriginal, System.String searchPattern, System.Boolean includeFiles, System.Boolean includeDirs, System.IO.SearchOption searchOption, System.Boolean checkHost) [0x00000] in <00000000000000000000000000000000>:0 
  at System.IO.Directory.GetDirectories (System.String path) [0x00000] in <00000000000000000000000000000000>:0 
  at ServerListingLocalUI.EditBackups (System.String fileDirectory, System.String fileName, System.Action`1[T] whatToDo) [0x00000] in <00000000000000000000000000000000>:0 
  at ServerListingLocalUI.<Delete>b__19_0 () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.AndroidJavaRunnable.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at PopupDialogUI.OnYesClick () [0x00000] in <00000000000000000000000000000000>:0 
  at Steamworks.ISteamMatchmakingRulesResponse+RulesFailedToRespond.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.Events.InvokableCall.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.Events.UnityEvent.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at System.UnhandledExceptionEventHandler.Invoke (System.Object sender, System.UnhandledExceptionEventArgs e) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.StandaloneInputModule.ReleaseMouse (UnityEngine.EventSystems.PointerEventData pointerEvent, UnityEngine.GameObject currentOverGo) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.StandaloneInputModule.ProcessMousePress (UnityEngine.EventSystems.PointerInputModule+MouseButtonEventData data) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.StandaloneInputModule.ProcessMouseEvent (System.Int32 id) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.StandaloneInputModule.Process () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.EventSystems.EventSystem.Update () [0x00000] in <00000000000000000000000000000000>:0 
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
UnityEngine.EventSystems.ExecuteEvents:Execute(GameObject, BaseEventData, EventFunction`1)
UnityEngine.EventSystems.StandaloneInputModule:ReleaseMouse(PointerEventData, GameObject)
UnityEngine.EventSystems.StandaloneInputModule:ProcessMousePress(MouseButtonEventData)
UnityEngine.EventSystems.StandaloneInputModule:ProcessMouseEvent(Int32)
UnityEngine.EventSystems.StandaloneInputModule:Process()
UnityEngine.EventSystems.EventSystem:Update()
 
(Filename: currently not available on il2cpp Line: -1)

[04:05:49.711] Nat introduction received with token = 40f05175-e4de-4946-86ce-3924db5f4d1d
EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Eco.Shared.Utils.Log:Debug(String, Object[])
Eco.Shared.Networking.NetworkClient:HandleCompleteUnconnectedDataMessage(UdpDataMessage&)
Eco.Shared.Networking.NetworkClient:HandleUnconnectedDataMessage(UdpDataMessage&)
Eco.Shared.Networking.Udp.UdpDataMessageHandler:Invoke(UdpDataMessage&)
Eco.Networking.ENet.ENetUdpPeer:OnRawDataReceived(Address&, IntPtr, Int32, Boolean&)
Eco.Networking.ENet.ENetUdpPeer:Intercept(Event&, Address&, IntPtr, Int32)
ENet.InterceptCallback:Invoke(Event&, Address&, IntPtr, Int32)
ENet.Host:Service(Int32, Event&)
Eco.Networking.ENet.ENetUdpPeer:Service(Int32)
<>c__DisplayClass63_0:<Initialize>b__0()
Steamworks.RulesFailedToRespond:Invoke()
System.Action`1:Invoke(T)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
Steamworks.RulesFailedToRespond:Invoke()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

[04:05:49.723] Nat punch received from 185.216.25.172:3000 with token = 40f05175-e4de-4946-86ce-3924db5f4d1d
EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Eco.Shared.Utils.Log:Debug(String, Object[])
Eco.Shared.Networking.NetworkClient:HandleCompleteUnconnectedDataMessage(UdpDataMessage&)
Eco.Shared.Networking.NetworkClient:HandleUnconnectedDataMessage(UdpDataMessage&)
Eco.Shared.Networking.Udp.UdpDataMessageHandler:Invoke(UdpDataMessage&)
Eco.Networking.ENet.ENetUdpPeer:OnRawDataReceived(Address&, IntPtr, Int32, Boolean&)
Eco.Networking.ENet.ENetUdpPeer:Intercept(Event&, Address&, IntPtr, Int32)
ENet.InterceptCallback:Invoke(Event&, Address&, IntPtr, Int32)
ENet.Host:Service(Int32, Event&)
Eco.Networking.ENet.ENetUdpPeer:Service(Int32)
<>c__DisplayClass63_0:<Initialize>b__0()
Steamworks.RulesFailedToRespond:Invoke()
System.Action`1:Invoke(T)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
Steamworks.RulesFailedToRespond:Invoke()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

[04:05:49.735] Connecting to 185.216.25.172:3000...

EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Eco.Shared.Utils.Log:WriteLine(LocString)
Eco.Shared.Networking.NetworkClient:ConnectInternal(IPEndPoint)
Eco.Shared.Networking.NetworkClient:ConnectRoundRobin()
Eco.Shared.Networking.NetworkClient:HandleNatIntroductionSuccess(IPEndPoint, String)
Eco.Shared.Networking.NetworkClient:HandleCompleteUnconnectedDataMessage(UdpDataMessage&)
Eco.Shared.Networking.NetworkClient:HandleUnconnectedDataMessage(UdpDataMessage&)
Eco.Shared.Networking.Udp.UdpDataMessageHandler:Invoke(UdpDataMessage&)
Eco.Networking.ENet.ENetUdpPeer:OnRawDataReceived(Address&, IntPtr, Int32, Boolean&)
Eco.Networking.ENet.ENetUdpPeer:Intercept(Event&, Address&, IntPtr, Int32)
ENet.InterceptCallback:Invoke(Event&, Address&, IntPtr, Int32)
ENet.Host:Service(Int32, Event&)
Eco.Networking.ENet.ENetUdpPeer:Service(Int32)
<>c__DisplayClass63_0:<Initialize>b__0()
Steamworks.RulesFailedToRespond:Invoke()
System.Action`1:Invoke(T)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
Steamworks.RulesFailedToRespond:Invoke()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

The referenced script on this Behaviour (Game Object 'LayerViewer') is missing! 
(Filename:  Line: 334)

Unable to open archive file: F:/Steam/steamapps/common/Eco/Eco_Data/StreamingAssets/desc.unity3d 
(Filename:  Line: 584)

Failed to read data for the AssetBundle 'StreamingAssets\desc.unity3d'. 
(Filename:  Line: 449)

[04:05:59.038] Missing sprite in MiscIconMap.
EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
ModBundleManager:SetupSprites()
<ReceiveModData>d__47:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

[04:05:59.250] Missing following objects: AirPollutionGeneratorObject, CountryMarkerObject, FedeartionMarkerObject, HomesteadMarkerObject, TownMarkerObject
EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogWarning(Object)
<BuildFromModKit>d__8:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<ReceiveModData>d__47:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

[04:06:00.726] Disconnected with error: Server encountered an exception:

<size=60.00%>Exception: NullReferenceException
Message:Object reference not set to an instance of an object.
Source:Eco.Gameplay

System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.Gameplay.Economy.Jobs.RecurringTransfer.get_SourceAccount()
   at Eco.Gameplay.Economy.Jobs.RecurringTransferManager.<>c__DisplayClass8_0.<TransfersForUser>b__0(RecurringTransfer x)
   at System.Linq.Utilities.<>c__DisplayClass1_0`1.<CombinePredicates>b__0(TSource x)
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at Eco.Core.Controller.ControllerManager.PackageValue(IMvcNetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageProperty(IMvcNetClient boundClient, IController controller, ISyncPropertyInfo propertyInfo)
Outer Exceptions:
Error during packaging property 'IncomeAndExpenses' of 'Eco.Gameplay.Players.User'</size>
EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogWarning(Object)
NetworkManager:OnClientDisconnected(String)
System.Action`1:Invoke(T)
Eco.Shared.Networking.NetworkClient:HandleDisconnect(String)
UnityEngine.Experimental.Audio.SampleFramesHandler:Invoke(AudioSampleProvider, UInt32)
Eco.Networking.ENet.ENetUdpClient:OnDisconnected(ENetConnection, DisconnectReason)
Eco.Networking.ENet.ENetUdpPeer:HandleEvent()
Eco.Networking.ENet.ENetUdpPeer:Service(Int32)
<>c__DisplayClass63_0:<Initialize>b__0()
Steamworks.RulesFailedToRespond:Invoke()
System.Action`1:Invoke(T)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
Steamworks.RulesFailedToRespond:Invoke()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

[04:06:00.744] Missing Connecting
EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
UI.ConnectionUI:GotoState(ConnectState)
UI.ConnectionUI:ManualUpdate()
UI.UIManager:Update()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

Unloading 7 Unused Serialized files (Serialized files now loaded: 0)

Unloading 31471 unused Assets to reduce memory usage. Loaded Objects now: 97479.
Total: 168.441500 ms (FindLiveObjects: 4.650100 ms CreateObjectMapping: 9.691400 ms MarkObjects: 134.799900 ms  DeleteObjects: 19.299600 ms)
```

Attached is the dump file.

[ClientUpdateException Exception 02020601.zip](https://github.com/StrangeLoopGames/EcoIssues/files/5912142/ClientUpdateException.Exception.02020601.zip)

**Comment by PixlRainbow** [comment]
experiencing same error as above, with the difference being that the crash did not occur until *after* the server auto-updated to 9.2.3 and we did not previously encounter any issues with wages.

**Comment by IcedForge** [comment]
Is this confirmed to be the case after updating to 9.2.4 As it contained further fixes towards reoccurring payments?
@PixlRainbow @Venopsis

**Comment by JNoppers** [comment]
This error has still occured for our server.

**Comment by devpikachu** [comment]
Can confirm, the error still occurs on 9.2.4.

**Comment by IcedForge** [comment]
Thank you for replying, can either of you upload your respective save files that are still having this issue for me to test with?

**Comment by devpikachu** [comment]
Sure thing, files attached as they were pulled from the server at the time of the incident.

https://we.tl/t-8uuQbTplb5

**Comment by IcedForge** [comment]
Thank you for following up with a save.
I can confirm it exploded on my end in 9.2.4 build 156
I can also confirm that this issue is fixed in 9.2.5 build 162
Is also working in 9.3
Will close this issue as it is confirmed working in next updates.
